lib/Service/FilesAppService.php

```diff
@@ -94,11 +98,13 @@ public function listAttachments(int $cardId): array {
 				$this->logger->debug('Unable to find node for share with ID ' . $share->getId());
 				return null;
 			}
+			$user = $this->userManager->get($this->userId);
+			$displayName = $user ? $user->getDisplayName() : '';
 			$attachment = new Attachment();
 			$attachment->setType('file');
 			$attachment->setId((int)$share->getId());
 			$attachment->setCardId($cardId);
-			$attachment->setCreatedBy($share->getSharedBy());
+			$attachment->setCreatedBy($displayName);
```

Comment:
The Entity classes like `Attachment` in deck allow to mark a field as resolvable to then later on fill it with a more detailed object if needed. This way we can pass both the user id (uid) as well as the display name to the frontend using a [User object](https://github.com/nextcloud/deck/blob/fc5c34897e601ac96f2b3c482791d075a35d831d/lib/Db/User.php) with something like this:

```php
$attachment->resolveRelation('createdBy', function ($userId) {
	$user = $this->userManager->get($userId);
	if ($user !== null) {
		return new User($user);
	}
	return $user;
});
```

Now the crucial point here is that we have two API endpoints where the result of this method may be used, the `AttachmentController` is the WebUI entrypoint and the `AttachmentApiController` is the public REST API which is for example used by the community android app. For the bug fixing part of this issue I'd say we should only change the result of the createdBy field to the Object to avoid changes in the API results. This would mean to resolve the relation only in https://github.com/nextcloud/deck/blob/199cccf86bd9386e7bb621e5bef724b02f5dee20/lib/Controller/AttachmentController.php#L44

For the REST API we should probably introduce a minor API version bump similar to https://github.com/nextcloud/deck/blob/199cccf86bd9386e7bb621e5bef724b02f5dee20/lib/Controller/AttachmentApiController.php#L47 and introduce a new 1.2 in https://github.com/nextcloud/deck/blob/199cccf86bd9386e7bb621e5bef724b02f5dee20/lib/Capabilities.php#L54

Reply:
Please check again. :)

Reply:
Maybe you can check the description above about the resolveRelation method and moving it to the AttachmentController again. Unfortunately it cannot just be put in the service class, as that would introduce a breaking API change.

If doing that as mentioned there should be a fallback for the older API versions and the version should be bumped.

Reply:
@juliushaertl Did you have a chance to check it again?